doc/help/kdb-record.md

```diff
@@ -0,0 +1,144 @@
+# kdb-record(1) -- Session recording for Elektra
+
+## SYNOPSIS
+
+`kdb record <COMMAND>`
+
+Where `<COMMAND>` is one of the following:
+
+- `start [session_name]`: start or resume a recording session.
+- `stop`: stop the active recording session.
+- `reset [session_name]`: remove all logged data from the recording session.
+- `delete [session_name]`: delete the recording session.
+- `mark [--meta meta_name] <key_name>`: marks a key to be used for validation.
+- `list`: list all available recording sessions.
+- `name`: prints the name of the currently active session.
+- `changes [session_name]`: shows a human-readable list of changes.
+- `export  [--replay|--diff] [session_name]`: creates a file with the changes in the session.
+- `import [--disable-validation] <file_name>`: imports files created by the `export` command.
+
+## DESCRIPTION
+
+The `kdb record` utility is used to manage session recordings.
+Only a single recording session can be active per user at a time.
+A user can have multiple non-active sessions.
+
+If a session is active, every change to the keys, values and meta-data stored in Elektra will be logged into that session.
+
+## ENVIRONMENT VARIABLES
+
+- `ELEKTRA_RECORD_ACTIVE_SESSION`: If this environment variable is set, the specified session will be the active session.
+
+## COMMON PARAMETERS
+
+Most commands allow specification of the name of the session they should work against.
+If no session name is provided, the active session will be used.
+
+A notable exception is the `start` command.
+If no session name is provided, it will create a random name.
+
+## COMMANDS
+
+### `start [session_name]`
+The `start` command is used to start a new or resume an existing session.
+If the environment variable `ELEKTRA_RECORD_ACTIVE_SESSION` is set, this command has no immediate effect.
+As soon as the environment variable is unset, the session started or resumed with this command will be active.
+
+This command will accept a single optional argument that is the name of the session.
+If the name is not provided, a random name will be generated.
+
+On success, the name of the session will be printed to `stdout`, and the return code is `0`.
+
+### `stop`
+The `stop` command is used to stop the active recording session.
+The session can then be resumed using the `start` command with the appropriate session name.
```

Comment:
"Resuming a session" needs more explanation. What happens if the KDB is in an incompatible state when the session is resumed. For example:

1. Start session `X`
2. Add key `user:/foo = 123`
3. Stop session `X`
4. Delete key `user:/foo`
5. Resume session `X`
6. Add key `user:/foo = 456`

Up until including 5 everything is fine. But at 6 the recording will record that `user:/foo` is newly added, when the recording already contains an entry that says "newly add `user:/foo`". Depending on the implementations this may be trivial to handle. However, I think maybe taking a different approach is easier (i.e. has fewer edge cases):

Instead of "resuming a session", I would say you always need to start a new session, but there is a kind command that concatenates finished sessions into a new one e.g. `kdb record cat`. That way the command can just check for inconsistencies and throw an error, if the sessions cannot be combined.

Reply:
I've added a new command `resume`. We can add those checks in this command.
I think this is more practical than starting a new session and concatenating. It enables us to throw an error before it's too late.